.infrastructure/helm-chart/todoapp/templates/hpa.yaml

```
@@ -0,0 +1,27 @@
+{{- if .Values.hpa.enabled }}
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ printf "%s-hpa" (include "todoapp.fullname" .) }}
+  namespace: {{ .Values.namespace }}
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: {{ include "todoapp.fullname" . }}
+  minReplicas: {{ .Values.hpa.minReplicas }}
+  maxReplicas: {{ .Values.hpa.maxReplicas }}
```

`minReplicas` and `maxReplicas` are read directly from `.Values.hpa.minReplicas`/`.Values.hpa.maxReplicas`. Make sure these fields exist and are integers in `values.yaml` (not strings). If you want safe defaults in the template, you can also use `default` or `required` functions to provide fallback values or fail loudly during `helm template`.